pkg/apis/storage/validation/validation.go

```diff
@@ -329,12 +329,25 @@ func validateCSINodeDriverNodeID(nodeID string, fldPath *field.Path) field.Error
 	return allErrs
 }
 
+// validateCSINodeDriverAllocatable tests if Allocatable in CSINodeDriver has valid volume limits.
+func validateCSINodeDriverAllocatable(a *storage.VolumeLimits, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	if a == nil || a.Count == nil {
+		return allErrs
+	}
+
+	allErrs = append(allErrs, apivalidation.ValidateNonnegativeField(int64(*a.Count), fldPath.Child("count"))...)
```

Comment:
yeah it was discussed in the KEP. @bsalamat expected this to be int32 since that should be enough to store volume limits.

Reply:
Do we want to validate maxint32 here too then?

Reply:
Since `*a.Count` is an  `int32`, I think we don't need to check if it's `> MaxInt32`, we only need to check if it's negative.

The worst thing that can happen is someone assigning `Count` with a truncated value (this will result in `Count == 1`, and we'd have the same problem if `Count` was an `int64`):

```
x := int64((math.MaxInt32 + 1) + (math.MaxInt32 + 1) + 1)
allocatable.Count = int32(x)
```